ui/app/components/pages/confirm-transaction-base/confirm-transaction-base.component.js

```diff
@@ -348,6 +400,29 @@ export default class ConfirmTransactionBase extends Component {
           })
       }
     })
+
+    setMetaMetricsSendCount(metaMetricsSendCount + 1)
+
+    if (onSubmit) {
+      Promise.resolve(onSubmit(txData))
```

Comment:
I don't think we need to wrap this in a `Promise` chain, as these all seem like sync calls. Could we write:

```js
try {
    onSubmit(...)
    this.setState(...)
} catch (e) {
    // ....
}
```